client/blocks/login/two-factor-authentication/waiting-notification-approval.jsx

```diff
@@ -37,23 +48,39 @@ class WaitingTwoFactorNotificationApproval extends Component {
 		} );
 	};
 
+	componentDidMount() {
+		this.props.startPollAppPushAuth();
+	}
+
+	componentWillUnmount() {
+		this.props.stopPollAppPushAuth();
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! this.props.pushSuccess && nextProps.pushSuccess ) {
+			this.props.onSuccess();
+		}
+	}
+
 	render() {
-		const { isSmsAuthSupported, translate } = this.props;
+		const {
+			isSmsAuthSupported,
+			pushSuccess,
+			translate,
+		} = this.props;
 
 		return (
 			<form>
 				<Card className="two-factor-authentication__push-notification-screen is-compact">
 					<p>
-						{ translate( 'We just sent a push notification to your WordPress mobile app. ' +
+						{ pushSuccess ? translate( 'Awesome, you are now logged in! One moment while we take you to homepage…' )
```

Comment:
@ranh 
Can you suggest an alternative for `Awesome, you are now logged in! One moment while we take you to homepage…` , it's displayed after the 2FA push is approved but before we completed the redirect.